src/main/drivers/bus_i2c_stm32f10x.c

```diff
@@ -86,12 +86,12 @@ const i2cHardware_t i2cHardware[I2CDEV_COUNT] = {
             I2CPINDEF(PF1,  GPIO_AF_I2C2),
         },
         .sdaPins = {
-            I2CPINDEF(PB11, GPIO_AF_I2C2),
```

Comment:
Looking at this, I am also wondering if this file should be renamed. :-)

Reply:
Any good name for f1 and f4 but not f3?  Or, just bus_i2c_stm32f4xx.c? 🤔

Reply:
Why do we have these split in this way anyway? And where are F7, H7?

Reply:
I think it's the history.

- F1 driver was implemented with stdperiphs (old `i2c_bus_stm32f10x.c`).
- F3 driver was implemented with stdperiphs (`i2c_bus_stm32f30x.c`). At this point, it wasn't not weird to have two different files for two different MCUs.
- When doing F4, someone found it is close enough to F1 (of course), so F4 driver parasitized on the old `i2c_bus_stm32f10x.c`.
- Some time after, the F1 driver was updated (I think it was done by @digitalentity) to use interrupts for state transitions rather than busy waiting (new `i2c_bus_stm32f10x.c`).
- Around the same time, F7 driver came along (by @sambas), which used HAL, hence `i2c_bus_hal.c`.
- I (@jflyper) did some code consolidation when I did pin configurability, which might have removed commonalities among these drivers (and thus further exaggerated diversities among them). 
- H7 driver came along, which naturally used `i2c_bus_hal.c`.
- G4 ditto.

Reply:
Hmm, how about combining `i2c_bus_stm32f10x.c` and `i2c_bus_stm32f30x.c` into `i2c_bus_stdperiph.c` then?